src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs

```diff
@@ -1126,7 +1131,10 @@ public override ClaimsPrincipal ValidateToken(string token, TokenValidationParam
             var samlToken = ValidateSignature(token, validationParameters);
             ValidateConditions(samlToken, validationParameters);
             var issuer = ValidateIssuer(samlToken.Issuer, samlToken, validationParameters);
-            ValidateTokenReplay(samlToken.Assertion.Conditions.NotOnOrAfter, token, validationParameters);
+
+            if (samlToken.Assertion.Conditions != null)
```

Comment:
>samlToken.Assertion [](start = 16, length = 19)

could this be null?

Reply:
From a derived SamlToken yes. I opened this issue to fix that.
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/1185